src/PKSim.Core/Mappers/IndividualToIndividualBuildingBlockMapper.cs

```diff
@@ -3,54 +3,85 @@
 using OSPSuite.Core.Domain;
 using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Domain.Formulas;
 using OSPSuite.Core.Domain.Services;
 using OSPSuite.Utility.Extensions;
 using PKSim.Assets;
 using PKSim.Core.Model;
 using PKSim.Core.Repositories;
+using PKSim.Core.Services;
+using IFormulaFactory = PKSim.Core.Model.IFormulaFactory;
+using ILazyLoadTask = OSPSuite.Core.Domain.Services.ILazyLoadTask;
 
 namespace PKSim.Core.Mappers
 {
-   public interface IIndividualToIndividualBuildingBlockMapper : IPathAndValueBuildingBlockMapper<Individual, IndividualBuildingBlock>
+   public interface IIndividualToIndividualBuildingBlockMapper : IPathAndValueBuildingBlockMapper<Individual, IndividualBuildingBlock, IndividualParameter>
    {
    }
 
    public class IndividualToIndividualBuildingBlockMapper : PathAndValueBuildingBlockMapper<Individual, IndividualBuildingBlock, IndividualParameter>, IIndividualToIndividualBuildingBlockMapper
    {
       private readonly IRepresentationInfoRepository _representationInfoRepository;
       private readonly ICalculationMethodCategoryRepository _calculationMethodCategoryRepository;
+      private readonly IParameterQuery _parameterQuery;
 
-      public IndividualToIndividualBuildingBlockMapper(IObjectBaseFactory objectBaseFactory, IEntityPathResolver entityPathResolver, IApplicationConfiguration applicationConfiguration,
-         ILazyLoadTask lazyLoadTask, IRepresentationInfoRepository representationInfoRepository, ICalculationMethodCategoryRepository calculationMethodCategoryRepository) : base(objectBaseFactory, entityPathResolver, applicationConfiguration, lazyLoadTask)
+      public IndividualToIndividualBuildingBlockMapper(IObjectBaseFactory objectBaseFactory,
+         IEntityPathResolver entityPathResolver,
+         IApplicationConfiguration applicationConfiguration,
+         ILazyLoadTask lazyLoadTask,
+         IRepresentationInfoRepository representationInfoRepository,
+         ICalculationMethodCategoryRepository calculationMethodCategoryRepository,
+         IFormulaFactory formulaFactory,
+         IParameterQuery parameterQuery) : base(objectBaseFactory, entityPathResolver, applicationConfiguration, lazyLoadTask, formulaFactory)
       {
          _representationInfoRepository = representationInfoRepository;
          _calculationMethodCategoryRepository = calculationMethodCategoryRepository;
+         _parameterQuery = parameterQuery;
+      }
+
+      protected override IFormula TemplateFormulaFor(IParameter parameter, IFormulaCache formulaCache, Individual individual)
+      {
+         bool isMetaDataForParameter(ParameterMetaData p) => p.BuildingBlockType == PKSimBuildingBlockType.Individual && string.Equals(p.ParameterName, parameter.Name);
+
+
+         //for individual, the CM to use depends on the CM available in the origin data as well as the container where the parameter resides.
+         var calculationMethods = individual.OriginData.AllCalculationMethods().AllNames();
```

Comment:
This is not very efficient because I am doing all of this in a loop but this seems to be very fast nonetheless